Source/DotSpatial.Data/Feature.cs

```diff
@@ -452,6 +452,15 @@ private static FeatureType FeatureTypeFromGeometryType(IGeometry geometry)
                 case OgcGeometryType.MultiPoint:
                     featureType = FeatureType.MultiPoint;
                     break;
+                case OgcGeometryType.GeometryCollection:
+                    IGeometryCollection geomCollection = geometry as IGeometryCollection;
+                    if (geomCollection != null)
+                    {
+                        if (geomCollection.NumGeometries > 0)
+                            featureType = FeatureTypeFromGeometryType(geomCollection[0]);

```

Comment:
Please make sure the featureType is the same for all the geometries in the collection. If not please assign FeatureType.Unspecified so functions assuming a certain type won't stumble over the differing geometries.